Source/monster.cpp

```diff
@@ -3648,7 +3648,10 @@ void MAI_Ranged(int i, int missile_type, BOOL special)
 				else
 					M_StartRAttack(i, missile_type, 4);
 			} else {
-				Monst->_mAnimData = Monst->MType->Anims[MA_STAND].Data[md];
+				fx = Monst->_lastx;
+				fy = Monst->_lasty;
+				md = GetDirection(Monst->_mx, Monst->_my, fx, fy);
+				M_CallWalk(i, md);

```

Comment:
Preferably the monster should not have decided to stand in the first place if this is not an appropriate action for it to take.

Do you have a save where this can be replicated?

Reply:
I'll get a save together with a monster specifically falling into this issue. I don't think it's bad for the AI to fall to this section of the code, the problem was Blizz made this section "just stand there." Ideally, you might do something a bit more than "Walk toward target" since this tends to induce a back and forth pacing (since it waffles between too close, and backed up to the spot that started all the trouble).

I'll get a save setup with this specific scenario though for ya in the morning.